src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -923,6 +957,763 @@ private FindRequestContext ConstructFindRequestContext(RestRequestContext contex
+        private void PerformDbInsertOperation(
+            IMiddlewareContext context,
+            object? parameters,
+            ISqlMetadataProvider sqlMetadataProvider,
+            MultipleCreateStructure multipleCreateStructure,
+            int nestingLevel)
+        {
+
+            if (multipleCreateStructure.InputMutParams is null || parameters is null)
+            {
+                throw new DataApiBuilderException(
+                        message: "The input for a multiple create mutation operation cannot be null.",
+                        statusCode: HttpStatusCode.BadRequest,
+                        subStatusCode: DataApiBuilderException.SubStatusCodes.BadRequest);
+            }
+
+            // For One - Many and Many - Many relationship types, the entire logic needs to be run for each element of the input.
+            // So, when the input parameters is of list type, we iterate over the list and run the logic for each element.
+            if (multipleCreateStructure.InputMutParams.GetType().GetGenericTypeDefinition() == typeof(List<>))
+            {
+                List<IDictionary<string, object?>> parsedInputParamList = (List<IDictionary<string, object?>>)multipleCreateStructure.InputMutParams;
+                List<IValueNode> paramList = (List<IValueNode>)parameters;
+                int idx = 0;
+
+                foreach (IDictionary<string, object?> parsedInputParam in parsedInputParamList)
+                {
+                    MultipleCreateStructure multipleCreateStrucutreForCurrentItem = new(entityName: multipleCreateStructure.EntityName,
+                                                                                        higherLevelEntityName: multipleCreateStructure.HigherLevelEntityName,
+                                                                                        inputMutParams: parsedInputParam,
+                                                                                        isLinkingTableInsertionRequired: multipleCreateStructure.IsLinkingTableInsertionRequired)
+                    {
+                        CurrentEntityParams = multipleCreateStructure.CurrentEntityParams,
+                        LinkingTableParams = multipleCreateStructure.LinkingTableParams
+                    };
+
+                    Dictionary<string, Dictionary<string, object?>> primaryKeysOfCreatedItems = new();
+                    IValueNode? nodeForCurrentInput = paramList[idx];
+                    if (nodeForCurrentInput is null)
+                    {
+                        throw new DataApiBuilderException(message: "Error when processing the mutation request",
+                                                          statusCode: HttpStatusCode.InternalServerError,
+                                                          subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
+                    }
+
```

Comment:
when would we encounter this scenario? is this a user caused error? a database triggered error?